modules/custom/cu_js_includes/cu_js_includes.module

```diff
@@ -0,0 +1,426 @@
+      'page callback' => 'drupal_get_form',
+      'file' => 'pages/cu_js_includes.include_form.inc',
+      'page arguments' => array('cu_js_includes_form', 4),
+      'access arguments' => $access_permission,
+      'type' => MENU_NORMAL_ITEM,
+    );
+  }
+
+  return $items;
+}
+
+/**
+ * Implements hook_menu_local_tasks_alter().
+ *
+ * Adds an action link for creating includes to the includes overview page.
+ */
+function cu_js_includes_menu_local_tasks_alter(&$data, $router_item, $root_path) {
+  // Add action link to 'admin/content/js-includes' page.
+  if ($root_path == 'admin/content/js-includes') {
+    $data['actions']['output'][] = array(
+      '#theme' => 'menu_local_action',
+      '#link' => array(
+        'title' => t('Add Includes'),
+        'href' => 'admin/content/js-includes/add',
+      ),
+    );
+  }
+}
+
+/**
+ * Implements hook_permission().
+ *
+ * @return array
+ *   List of permissions.
+ */
+function cu_js_includes_permission() {
+  $permissions['administer js includes'] = array(
+    'title' => t('Administer JS Includes'),
+  );
+  $permissions['view js includes'] = array(
+    'title' => t('View JS Includes'),
+  );
+  $permissions['edit js includes'] = array(
+    'title' => t('Edit JS Includes'),
+  );
+  $permissions['delete js includes'] = array(
+    'title' => t('Delete JS Includes'),
+  );
+
+  return $permissions;
+}
+
+/**
+ * Implements hook_theme().
+ *
+ * Create a theme template for each embed type.
+ *
+ * @param array $existing
+ *   An array of existing theme definitions.
+ * @param string $type
+ *   Whether a theme, module, etc. is being processed.
+ * @param string $theme
+ *   Name of the thing being processed.
+ * @param string $path
+ *   The directory path of the theme or module.
+ *
+ * @return array
+ *   List of new theme definitions.
+ */
+function cu_js_includes_theme($existing, $type, $theme, $path) {
+  $include_types = cu_js_includes_get_include_types();
+  $templates = array();
+
+  foreach ($include_types as $key => $include_type) {
+    $templates['cu_js_includes_' . $key] = array(
+      'variables' => array('data' => NULL),
+      'path' => $include_type['path'],
+      'template' => $include_type['template'],
+    );
+  }
+
+  return $templates;
+}
+
+/**
+ * Get all includes from database.
+ *
+ * @return array
+ *   A list of all includes with correct headers for table display.
+ */
+function cu_js_includes_get_includes($conditions = array(), $sortable_header = NULL) {
+
+  // If headers sent in, then add sortable.
+  if ($sortable_header) {
+    $query = db_select('cu_js_includes', 'tp')->extend('TableSort');
+    $query->orderByHeader($sortable_header);
+  }
+  else {
+    $query = db_select('cu_js_includes', 'tp');
+  }
+
+  $query->fields('tp');
+
+  // Loop through options array and add conditions.
+  // Conditions have to match the db columns.
+  foreach ($conditions as $key => $condition) {
+    $query->condition($key, $condition, '=');
```

Comment:
Still getting an error for 'statuspage'. I have created a couple status page includes and these get passed tot he database as an `array('id' => [ids])`.

When `$condition` is an array the `=` operator throws off the database query. When I just removed the third argument altogether it works fine. If there is no third argument passed in the operator defaults to '=' for a scalar value and 'IN' for an array.

Reply:
Can you describe what steps you're doing and where the error shows up?

Reply:
Sure thing:
1. I have created 2 status page includes. (they show up in the list in the overview page -> /admin/content/js-includes)
2. I try to filter the includes in the list by selecting 'Statuspage' in the 'Include Type' select list.
3. The next page loaded in the browser matches this picture:
    - <img width="493" alt="Screen Shot 2020-10-06 at 1 09 12 PM" src="https://user-images.githubusercontent.com/23508839/95248957-3eebc780-07d5-11eb-956d-74dbdf01a181.png">

After this, using the debugger, I found that there is an error after calling `$query->execute()->fetchAllAssoc('id');` on line 198 with the message:
> SQLSTATE[21000]: Cardinality violation: 1241 Operand should contain 1 column(s)

Reply:
After poking around a bit more with the debugger found that the condition set, in this case where the `$condition` inside the loop on lines 194-6 is setting the query condition to 'id' = array(int => int, int => int).